File: GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
        for(int i = 0; i < strs.length; i++){
            char[] val = strs[i].toCharArray();
            Arrays.sort(val);
            String values = new String(val);
            if(!map.containsKey(values)){
                map.put(values, new ArrayList<String>());
            }
            map.get(values).add(strs[i]);
        }
        return new ArrayList<>(map.values());
    }
}

File: GroupAnagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        map = defaultdict(list)
        for i in strs:
            temp = i
            temp = "".join(sorted(temp))
            map[temp].append(i)
        return map.values()

        